Sources_v3/StreamChatUI/ChatChannel/ChatChannelRouter.swift

```diff
@@ -0,0 +1,35 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import UIKit
+
+open class ChatChannelRouter<ExtraData: UIExtraDataTypes>: ChatRouter<ChatChannelVC<ExtraData>> {
+    open func showMessageActionsPopUp(
+        messageContentFrame: CGRect,
+        messageData: _ChatMessageGroupPart<ExtraData>,
+        messageController: _ChatMessageController<ExtraData>,
```

Comment:
First I was thinking that the routers should have access to `client` but this solution is much better 👍